docs/jep/12049-type-annotations.md

```diff
@@ -219,33 +219,41 @@ Despite the familiarity of numpy's API choices, the `ndarray` / `NDArray` / `Arr
 
 Python itself is slowly moving to a world of unifying instance and annotation types; for example, with class-level `getitem` support in Python 3.9, classes like `typing.Dict`, `typing.List`, `typing.Tuple`, etc. are now [deprecated](https://peps.python.org/pep-0585/#implementation) in favor of their true-typed counterparts, `dict`, `list`, and `tuple`.
 
-With this in mind, JAX could instead choose to use its (eventual) `jax.Array` type directly for both annotation and instance checks. For handling tracers, tracers could be supported using a construct like the following:
+With this in mind, JAX could instead choose to use its (eventual) `jax.Array` type directly for both annotation and instance checks. For handling tracers within type annotations, we could use a construct like the following:
 ```python
 if TYPE_CHECKING:
   Array = Union[jax.Array, jax.Tracer]
 else:
-  Array = jax.Array
+  Array = jax._src.array.Array
```

Comment:
Nit: the `TYPE_CHECKING` branch hasn't been updated.

Reply:
Thanks!